### ISSUE TITLE
`xsd:simpleType` properties are resolved to non existing classes during code generation

### ISSUE THREAD

**Issue by ademarco** [phpro/soap-client#218]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 0.7.11

#### Summary

When generating code `xsd:simpleType`properties are resolved to a non existing type class instead of being resolved to their actual base type.

#### Current behavior

Take the current XSD snippet:

```xml
    <xsd:simpleType name="languageCode">
        <xsd:restriction base="xsd:string">
            <xsd:enumeration value="ML"/>
            <xsd:enumeration value="EN"/>
            <xsd:enumeration value="FR"/>
            <xsd:enumeration value="DE"/>
        </xsd:restriction>
    </xsd:simpleType>
```

This will generate the following property in the type classes that use that property:

```php
namespace MyLibrary\Service\Type;

class Document
{
    /**
     * @var MyLibrary\Service\Type\LanguageCode
     */
    protected $language;
}
```

The `MyLibrary\Service\Type\LanguageCode` class is not generated as it's not a complex type and only complex types are collected for generation in `SoapClient::getSoapTypes()`.

#### How to reproduce

You can reproduce by checking out [this commit](https://github.com/openeuropa/epoetry-client/commit/a4a708043e6c8212ae99190f9c7edc05a37c024e) of our library and run a code generation.

You can see also the bug at [this line](https://github.com/openeuropa/epoetry-client/commit/a4a708043e6c8212ae99190f9c7edc05a37c024e#diff-2a0a2d15bfbb4fe8055397f64a512023R11), for example.

#### Expected behavior

The expected behavior is to fallback to the primitive type the element refers to in its XSD definition, meaning, given:

```xml
    <xsd:simpleType name="languageCode">
        <xsd:restriction base="xsd:string">
        ...
```

We should get:

```php
namespace MyLibrary\Service\Type;

class Document
{
    /**
     * @var string
     */
    protected $language;
}
```

**Comment by veewee** [comment]
Thanks for reporting.

The problem with enum types is that we don't get the base type from ext-soap. With other words: we won't be able to provide that language is a string and not an int as long as we are relying on ext-soap.

I agree that we could add a mapping to the normalizer for other main simpleTypes like `base64Binary` etc.

So about your question in general: It is not something we can fix at this point.

**Comment by ademarco** [comment]
About:

> we won't be able to provide that language is a string and not an int as long as we are relying on ext-soap.

Not sure I understand, isn't that what we get from `$soapTypes = $this->__getTypes();` (see below)? 

![image](https://user-images.githubusercontent.com/153362/51919341-822c4b00-23e3-11e9-9bfe-1aac527403a4.png)

Isn't that derived by the following XSD?

```yml
    <xsd:simpleType name="languageCode">
        <xsd:restriction base="xsd:string">
        ...
```

**Comment by veewee** [comment]
Thanks for adding the screenshot. That is the first time I see those simple type maps. We could definitely do something with that! 

I'll have to run some additional test on how it behaves (e.g. duplicate enum definitions in different xml namespace) and how we should deal with this information.

We could simply convert those types to their base type, bit then the information of the simpletype is lost and this makes it harder to e.g. autogenerated enum types in the future.

**Comment by veewee** [comment]
Hello @ademarco 

I investigated the issue and added a fix to the new soap engine branch which will be released next.
Here is the commit:
https://github.com/phpro/soap-client/pull/197/commits/eccaa7b18240847faa0caebbc53a4892a48b7e6c

Feel free to test it out.

**Comment by veewee** [comment]
Fixed in version 1.0.0
